changedetectionio/fetch_site_status.py

```diff
@@ -276,11 +277,28 @@ def run(self, uuid):
                 else:
                     logging.debug("check_unique_lines: UUID {} had unique content".format(uuid))
 
+            diff_filters = {
+                "add": watch.get('trigger_on_add', True),
+                "del": watch.get('trigger_on_del', True),
+            }
+
+            if False in diff_filters.values(): # if we are supposed to filter any diff types
+                print(">>>>>>>>>>>> Filtering diffs")
+                # get the diff types present in the watch
+                diff_types = watch.get_diff_types(str(stripped_text_from_html))
+                
+                # for each diff type, if the filter setting is false, and the diff type is present, then set the changed_detected to false
+                for diff_type in diff_types:
+                    if not diff_filters[diff_type] and diff_types[diff_type]:
+                        changed_detected = False
+                        break # we only need to check one diff type
+
         # Always record the new checksum
         update_obj["previous_md5"] = fetched_md5
 
         # On the first run of a site, watch['previous_md5'] will be None, set it the current one.
         if not watch.get('previous_md5'):
             watch['previous_md5'] = fetched_md5
+            watch['previous_text'] = str(stripped_text_from_html)
```

Comment:
compare this with the previous in `.history` i think is better (if theres a setting enabled that requires it)

Reply:
@bwees hmm, but people will also setup like the CSS Filter, so they only want to know when certain parts of a website change (or have additions or removals)

I still think that this feature should not change this, it would be too confusing otherwise (related to filters)

but relating to triggers... hmmm

yes so one example would be, they only want to trigger
- when text is removed
- when text contains "xyz" 

but that should work by examining the last snapshot anyway? (without your custom snapshot saving)

Reply:
I am pretty sure that should not change because stripped_text_from_html is after the CSS extraction. As for the second part of your question, I am not sure how that would be affected. I think you would then have to make a hierarchy of tests that could be reordered in order to provide that functionality. (I feel that it is outside of the scope of this feature anyway) The case you provide would work if both of those conditions are true but not either/or.